spec/views/post_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+RSpec.describe 'Post', type: :feature do
+  before(:all) do
+    @user = User.create(name: 'Yunus', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.',
+                        posts_counter: 0)
+    @post = Post.create(author: @user, title: 'new post', text: 'I love posting', comments_counter: 0,
+                        likes_counter: 0)
+    Post.create(author: @user, title: 'new post', text: 'I love posting', comments_counter: 0, likes_counter: 0)
+    Post.create(author: @user, title: 'new post', text: 'Post 16', comments_counter: 0, likes_counter: 0)
+    Post.create(author: @user, title: 'new post', text: 'Post 2', comments_counter: 0, likes_counter: 0)
+    Comment.create(author: @user, post: @post, text: 'I love comments')
+  end
+
+  describe 'index' do
+    it 'shows the users profile picture' do
+      visit "/users/#{@user.id}/posts"
+      img = page.find('img')
+      expect(img['src']).to have_content('https://unsplash.com/photos/F_-0BxGuVvo')
+    end
+
+    it 'shows the users username' do
+      visit "/users/#{@user.id}/posts"
+      expect(page).to have_content('Yunus')
+    end
+
+    it 'shows the users posts counter' do
+      visit "/users/#{@user.id}/posts"
+      expect(page).to have_content('Number of posts: 4')
+    end
+
+    it 'shows the title of a post' do
+      visit "/users/#{@user.id}/posts"
```

Comment:
- [ ] some of your tests are failing please make sure to correct them so that they're passing 
![Screenshot 2022-12-06 at 14 54 50](https://user-images.githubusercontent.com/61790587/205931119-084ce0d4-11bd-450c-b4b1-c4359892d19a.png)

Reply:
@youmari those tests were for the controller when database cleaner is used the data we were comparing to doesn't exist why it happened? and for n+1 problems? adding it didn't solve it ...kindly let me if you have any suggestions on how to do it

Reply:
@youmari for n+1 problem add `.includes(:comments)` it will solve the prob and for tests you should fix them so they can pass

Reply:
using the `.includes(:comments)
![fail screenshot](https://user-images.githubusercontent.com/32224137/205947339-7369b19c-3ee3-4b43-abdd-e9f542e89c86.PNG)

I have fixed the test though